zulip/integrations/bridge_with_matrix/matrix_bridge.py

```diff
@@ -1,28 +1,40 @@
 #!/usr/bin/env python3
 import argparse
+import asyncio
 import configparser
 import logging
 import os
 import re
 import signal
 import sys
 import traceback
+import urllib.request
 from collections import OrderedDict
-from types import FrameType
-from typing import Any, Callable, Dict, Optional
-
-from matrix_client.client import MatrixClient
-from matrix_client.errors import MatrixRequestError
-from requests.exceptions import MissingSchema
+from concurrent.futures import ThreadPoolExecutor
+from io import BytesIO
+from typing import Any, Dict, List, Match, Optional, Set, Tuple, Type, Union
+
+import magic
+import magic.compat
+import nio
+from nio.responses import (
+    DownloadError,
+    DownloadResponse,
+    ErrorResponse,
+    JoinError,
+    JoinResponse,
+    LoginError,
+    LoginResponse,
+    Response,
+    SyncError,
+    SyncResponse,
+)
 
 import zulip
 
-GENERAL_NETWORK_USERNAME_REGEX = "@_?[a-zA-Z0-9]+_([a-zA-Z0-9-_]+):[a-zA-Z0-9.]+"
-MATRIX_USERNAME_REGEX = "@([a-zA-Z0-9-_]+):matrix.org"
-
 # change these templates to change the format of displayed message
-ZULIP_MESSAGE_TEMPLATE = "**{username}**: {message}"
-MATRIX_MESSAGE_TEMPLATE = "<{username}> {message}"
+ZULIP_MESSAGE_TEMPLATE: str = "**{username}** [{uid}]: {message}"
+MATRIX_MESSAGE_TEMPLATE: str = "<{username} ({uid})> {message}"
```

Comment:
Why is sender_id necessary in the template? Same question for the Zulip template.

Reply:
I thought it might be good to resolve name collisions when multiple people have the same user name. Because you cannot see their avatars and thus might be confused or get tricked.